tests/rwlock.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <climits>
+#include <condition_variable>
+#include <mutex>
+
+class ReaderWriterLock {
```

Comment:
Hm, can you explain a bit why we need to implement a read write lock?

Reply:
In order to decide how to treat a rpc, discard or delay, proxy nodes have to get the current network config.

How they get the config? 

They share the NetworkConfig with a shared_ptr<NetworkConfig> and read the NetworkConfig to get the current network config. At the same time, the user may modify the network config and we need to protect the NetworkConfig with a lock.

why rwlock?

Suppose we have 5 proxy nodes and there are 5 rpc messages reach to these proxy nodes at the same time. Only one proxy node could forward the rpc when we simply use a std::mutex. But I hope these 5 proxy nodes could forward the rpc at the same time.

Reply:
We have supported C++17 https://github.com/purecpp-org/raftcpp/blob/master/CMakeLists.txt#L3

RWlock is supported in std. It's better to use it in C++ std way.

Reply:
Yes, std::shared_mutex from c++17.
https://en.cppreference.com/w/cpp/thread/shared_mutex

Reply:
> We have supported C++17 https://github.com/purecpp-org/raftcpp/blob/master/CMakeLists.txt#L3
> 
> RWlock is supported in std. It's better to use it in C++ std way.

I agree it!